vllm/v1/executor/ray_distributed_executor.py

```diff
@@ -55,10 +145,22 @@ def execute_model(
 
         refs = self.forward_dag.execute(scheduler_output)  # type: ignore
 
-        # When PP is not used, we block here until the result is available.
-        if self.max_concurrent_batches == 1:
-            return refs[0].get()
+        if not self.has_connector:
```

Comment:
I think the changes below here in this file should be all that's needed in this PR (apart from moving those other two functions so that they can be shared).